base/error.jl

```diff
@@ -65,9 +65,14 @@ struct InterpreterIP
     mod::Union{Module,Nothing}
 end
 
+const _previous_bt = Array{Any,1}(undef, 2)
+
 # convert dual arrays (raw bt buffer, array of GC managed values) to a single
 # array of locations
 function _reformat_bt(bt, bt2)
+    global _previous_bt
+    _previous_bt[1] = bt
+    _previous_bt[2] = bt2
```

Comment:
@vtjnash I think it's these lines which have caused the backtrace tests to pass again in your latest change because this code runs in the innermost frame of `g29695`.

In trying to debug this, I've quite consistently found that any code which I insert in the innermost frame to observe state causes the tests to pass. Surprisingly enough this was true for the C code as well as code on the julia side.

OTOH, it seems fine to run whatever code you like once the pile of recursive calls has returned.